src/clims/services/extensible.py

```diff
@@ -77,11 +87,19 @@ def register(self, plugin_reg, extensible_base):
             property_types.append(prop_type)
 
         name = "{}.{}".format(extensible_base.__module__, extensible_base.__name__)
-        self._implementations[name] = extensible_base
         extensible_type = self._register_model(
             name, 'default', plugin_reg, property_types=property_types)
         return extensible_type
 
+    def unregister_model(self, extensible_base, plugin):
+        """
+        This is for testing purposes only!
```

Comment:
I would prefer that code that's just for tests is kept in the tests only. I believe that keeping only what's required in the API makes it simpler to manage, e.g. in terms of knowing what requires tests.